File: net/ChainOfResponsability/ChainConsole/Chain/ChainBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChainConsole.Chain
{
    public abstract class ChainBase
    {
        private bool HasNext
        {
            get
            {
                return Next != null;
            }
        }

        public ChainBase Next { get; set; }

        internal void MoveOn(string data)
        {
            if (HasNext)
                Next.Process(data);
        }

        public abstract void Process(string data);
    }
}

File: java/ChainOfResponsability/src/ChainBase.java

public abstract class ChainBase {
	private ChainBase next_element;
	
	public ChainBase getNext() {
		return next_element;
	}
	
	public void setNext(ChainBase chain) {
		next_element = chain;
	}
	
	public void MoveOn(String data) {
		if (next_element != null) {
			try {
				next_element.Process(data);
			} catch (Exception e) {
				System.out.print("Error");
			}
		}
	}
	
	public abstract void Process(String data) throws Exception;
	
}